File: JavaScript/1370-Increasing Decreasing String.js
/**
 * @param {string} s
 * @return {string}
 */
var sortString = function(s) {
    let hmap = new Map();
    for (let char of s){
        hmap.set(char, (hmap.get(char)||0)+1);
    }
    let res = "";
    while (res.length < s.length){
        for (let char of 'abcdefghijklmnopqrstuvwxyz'){
            if (hmap.get(char)){
                res += char;
                hmap.set(char, hmap.get(char)-1);
            }
        }
        
        for (let char of 'zyxwvutsrqponmlkjihgfedcba'){
            if (hmap.get(char)){
                res += char;
                hmap.set(char, hmap.get(char)-1);
            }
        }
    }
    return res;
};

File: 1370-Increasing Decreasing String.py
from collections import Counter
class Solution(object):
    def sortString(self, s):
        """
        :type s: str
        :rtype: str
        """
        hmap = Counter(s)
        res = ""
        while len(res) < len(s):
            for char in 'abcdefghijklmnopqrstuvwxyz':
                if hmap[char]:
                    res += char
                    hmap[char] -= 1
                    
            for char in 'zyxwvutsrqponmlkjihgfedcba':
                if hmap[char]:
                    res += char
                    hmap[char] -= 1
        
        return res